general/glossary.md

```diff
@@ -0,0 +1,52 @@
+# Glossary
+
+This documents aims at providing a common lexicon for terms used to refer to artifacts.
+
+### Motivation
+
+Throughout the documentation there are several terms which are often abused because we either have no alternative or there's not enough consensus on what's their canonical name.
+
+This leads to often ambiguous documentation because one term is used for different things or because we use different terms for the same artifact/object.
+
+If this document succeeds, we should end up having a documentation where each object/artifact has one single term to designate it. In other words, [Wittgenstein](https://en.wikipedia.org/wiki/Ludwig_Wittgenstein) should be proud of us (see [Tractatus Logico-Philosophicus](https://en.wikipedia.org/wiki/Tractatus_Logico-Philosophicus)).
+
+Note that we also include an [appendix section](#appendix-trade-marks--product-names) listing the canonical spelling and casing for used trade marks or product names.
+
+## Technical Terms
+
+| Term              | Description                                                                       |
+| ----------------- | --------------------------------------------------------------------------------- |
+| AMD module        | A JavaScript AMD module (the `.js` file or its logical representation at runtime) |
+| CommonJS module   | A JavaScript CommonJS module (the `.js` files used in node)                       |
+| ES6+ module       | EcmaScript 6 or higher code (i.e.: `import from` syntax)                          |
+| Harmony module    | Same as ES6+ module (better use _ES6+ module_ instead)                            |
+| Java package      | A Java package as defined by the language                                         |
+| JavaScript module | A generic JavaScript module in any format (AMD, CommonJS, ...)                    |
+| Module            | A _JavaScript module_ (only to be used when there's no ambiguity)                 |
+| npm package       | A JavaScript package (those described by a `package.json` file)                   |
+| OSGi bundle       | The JAR file deployed to DXP or its logical representation at runtime             |
```

Comment:
I think we should define "OSGi module" as well, seeing as that's what we've got in liferay-portal under `modules/`.